writeengine/wrapper/we_colop.cpp

```diff
@@ -1738,6 +1743,12 @@ int ColumnOp::writeRow(Column& curCol, uint64_t totalRow, const RID* rowIdArray,
         }
 
         // This is the write stuff
+        if (oldValArray)
```

Comment:
I would move this block into a separate inline function to reduce boilerplate but this is not mandatory.